pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -597,3 +545,86 @@ func (l *AccessCredentialManager) HandleQemuAgentAccessCredentials(vmi *v1.Virtu
 
 	return nil
 }
+
+type accessCredentialsInfo struct {
+	// secret name mapped to authorized_keys in that secret
+	secretMap map[string][]string
+	// filepath mapped to secretNames
+	userSSHMap map[string][]string
+	// maps users to passwords
+	userPasswordMap map[string]string
+}
+
+func (a *accessCredentialsInfo) addAccessCredential(accessCred *v1.AccessCredential) error {
+	secretName := getSecret(accessCred)
+	if secretName == "" {
+		return nil
+	}
+
+	secretDir := getSecretDir(secretName)
+	files, err := os.ReadDir(secretDir)
+	if err != nil {
+		return fmt.Errorf("error encountered reading secrets file list from base directory %s: %w", secretDir, err)
+	}
+
+	if isSSHPublicKey(accessCred) {
+		for _, user := range accessCred.SSHPublicKey.PropagationMethod.QemuGuestAgent.Users {
+			secrets, ok := a.userSSHMap[user]
+			if !ok {
```

Comment:
nit as this isn't new code but you should always be able to `a.userSSHMap[user] = append(secrets, secretName)` here regardless of the value of `ok`?

https://go.dev/play/p/h-FeG2MHZ1N

Reply:
Right. I've simplified this.